platform/podjs/bin/genPodjs.js

```diff
@@ -0,0 +1,75 @@
+const MagicString = require("magic-string");
+const fs = require("fs");
+
+const yargs = require("yargs");
+
+const argv = yargs
+    .option("podjs", {
+        alias: "podjs",
+        description: "Tell the path of podjs file",
+        type: "string",
+    })
+    .option("manifestJson", {
+        alias: "manifestJson",
+        description: "Tell the path to manifest json file",
+        type: "string",
+    }).argv;
+
+const podjs = argv.podjs;
+const manifestJsonPath = argv.manifestJson;
+
+function replaceManifestData(code, manifestData) {
```

Comment:
Instead of replacing `window.manifestData` in `pod.js`, I would simply prepend some code at the top that sets it:

```
window.manifestData = ${manifestData};
```

Or - for this you'd have to modify the code in _podjs_ further, we make it non-global:

```
const manifestData = ${manifestData};
```

Reply:
Ah, I see that you are now reading `window.manifestData` only once - in that case replacing is OK. But if we manage to just prepend the code, we could make this script a whole lot shorter :D If it was a bash script, it'd just be something like:

```
echo "window.manifestData = $(cat manifest.json);" > pod.js
cat original-pod.js >> pod.js
```